### PR TITLE
aggregation/txmetrics: include transaction outcome

### LINKED ISSUE #4041: Add event.outcome to metrics aggregation
See https://github.com/elastic/apm/pull/299

We'll be adding `event.outcome` to transaction docs, which (per https://www.elastic.co/guide/en/ecs/current/ecs-allowed-values-event-outcome.html) has a limited set of values: success, failure, unknown, (unset)

This field will be used for error rates. So that we can use aggregated metrics docs for the calculations, we will need to include this field in transaction metrics docs, which also implies that it needs to be included in the aggregation key. This should have no or little impact on series cardinality, as `event.outcome` will typically be subsumed by `transaction.result`.

### PR DESCRIPTION
**Pull request by axw**
## Motivation/summary

Add `Transaction.Outcome` to aggregation key, and add `event.outcome` to the metrics docs produced.

## Checklist

- [x] I have signed the [Contributor License Agreement](https://www.elastic.co/contributor-agreement/).
- [x] I have updated [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/master/CHANGELOG.asciidoc)

I have considered changes for:
~- [ ] documentation~
~- [ ] logging (add log lines, choose appropriate log selector, etc.)~
~- [ ] metrics and monitoring (create issue for Kibana team to add metrics to visualizations, e.g. [Kibana#44001](https://github.com/elastic/kibana/issues/44001))~
- [x] automated tests (add tests for the code changes, all [**unit** tests](https://github.com/elastic/apm-server/blob/master/TESTING.md) pass locally)
~- [ ] telemetry~
~- [ ] Elasticsearch Service (https://cloud.elastic.co)~
~- [ ] Elastic Cloud Enterprise (https://www.elastic.co/products/ece)~
~- [ ] Elastic Cloud on Kubernetes (https://www.elastic.co/elastic-cloud-kubernetes)~

## How to test these changes

1. Start apm-server with transaction metrics aggregation enabled
2. Send some transactions with differing outcomes (e.g. using apm-agent-python)
3. Observe multiple metrics are produced with the same outcome values

## Related issues

Closes #4041

### CODE REVIEW

File: x-pack/apm-server/aggregation/txmetrics/aggregator_test.go [hunk 1]
```diff
@@ -341,6 +341,93 @@ func testHDRHistogramSignificantFigures(t *testing.T, sigfigs int) {
...
+		&input.Metadata.Service.Agent.Name,
+		&input.Metadata.Service.Environment,
+		&input.Metadata.Service.Name,
+		&input.Metadata.Service.Version,
+		&input.Metadata.System.Container.ID,
+		&input.Metadata.System.Kubernetes.PodName,
+	}
+
+	var expected []model.Metricset
+	expectCount := func(expectedCount int64) {
```
Comment (jalvz) [inline x-pack/apm-server/aggregation/txmetrics/aggregator_test.go:372]: this function name makes me thing that there is an assertion inside, but it just adds an item to a list? So `addToExpected` or something like that perhaps?

**Review by jalvz — APPROVED** [review]

File: x-pack/apm-server/aggregation/txmetrics/aggregator_test.go (hunk 1 above)
Comment (axw) [reply x-pack/apm-server/aggregation/txmetrics/aggregator_test.go:372]: Renamed to `addExpectedCount`, happy to iterate if that's still unclear.

**Comment by axw** [comment]
Tests failed due to a data race in unrelated code, merging.

**Comment by simitt** [comment]
Successfully tested;
* started APM Server with aggregation enabled
* sent transactions with same/different `service.name` and same/different `http.response.status_code` and ensured expected amount of histogram metrics was created

**Pull request merged**